crates/proof-of-sql/src/sql/postprocessing/evaluate.rs

```diff
@@ -0,0 +1,267 @@
+            Literal::Timestamp(its) => Ok(OwnedColumn::TimestampTZ(
+                its.timeunit,
+                its.timezone,
+                vec![its.timestamp.timestamp(); len],
+            )),
+        }
+    }
+
+    fn visit_unary_expr(
+        &self,
+        op: UnaryOperator,
+        expr: &Expression,
+    ) -> PostprocessingResult<OwnedColumn<S>> {
+        let column = self.visit_expr(expr)?;
+        match op {
+            UnaryOperator::Not => {
+                if column.column_type() == ColumnType::Boolean {
+                    // We can unwrap here because we know the column is boolean
+                    Ok(OwnedColumn::Boolean(
+                        column
+                            .as_boolean()
+                            .unwrap()
+                            .into_iter()
+                            .map(|b| !b)
+                            .collect(),
+                    ))
+                } else {
+                    Err(PostprocessingError::UnaryOperationInvalidColumnType {
+                        operator: UnaryOperator::Not,
+                        operand_type: column.column_type(),
+                    })
+                }
+            }
+        }
+    }
+
+    fn visit_binary_expr(
+        &self,
+        op: BinaryOperator,
+        left: &Expression,
+        right: &Expression,
+    ) -> PostprocessingResult<OwnedColumn<S>> {
+        let left = self.visit_expr(left)?;
+        let right = self.visit_expr(right)?;
+        let length = left.len();
+        assert_eq!(length, right.len());
+        check_dtypes(left.column_type(), right.column_type(), op)?;
+        match op {
+            BinaryOperator::And => Ok(OwnedColumn::Boolean(
+                left.as_boolean()
+                    .unwrap()
+                    .into_iter()
+                    .zip(right.as_boolean().unwrap().into_iter())
+                    .map(|(l, r)| *l && *r)
+                    .collect(),
+            )),
+            BinaryOperator::Or => Ok(OwnedColumn::Boolean(
+                left.as_boolean()
+                    .unwrap()
+                    .into_iter()
+                    .zip(right.as_boolean().unwrap().into_iter())
+                    .map(|(l, r)| *l || *r)
+                    .collect(),
+            )),
+            BinaryOperator::Equal
+            | BinaryOperator::GreaterThanOrEqual
+            | BinaryOperator::LessThanOrEqual => {
+                let lhs_scale = left.column_type().scale().unwrap_or(0);
+                let rhs_scale = right.column_type().scale().unwrap_or(0);
+                let max_scale = lhs_scale.max(rhs_scale);
+                let lhs_upscale_factor =
+                    scale_scalar(S::ONE, max_scale - lhs_scale).expect("Invalid scale factor");
+                let rhs_upscale_factor =
+                    scale_scalar(S::ONE, max_scale - rhs_scale).expect("Invalid scale factor");
```

Comment:
these ```.expects``` seem like they are reporting a failure, but it seems like expect should be used when we are 100% certain that this cannot fail, so a message explaining why the failure cannot occur seems more appropriate